lib/app/modules/smart/token_refresh_sequence.rb

```diff
@@ -144,7 +146,9 @@ def validate_and_save_refresh_response(token_response)
         end
       end
 
-      test 'Server successfully refreshes the access token when optional scope parameter omitted.' do
+      REFRESH_WITHOUT_SCOPE_PARAMETER_TEST =
+        'Server successfully refreshes the access token when optional scope parameter omitted.'
+      test REFRESH_WITHOUT_SCOPE_PARAMETER_TEST do
```

Comment:
This is interesting, but it seems like we are encouraging a pattern of using a test "key" as well as a "name"/"short description" but not enforcing it.  If we feel strongly this is the right approach, then we could update the DSL to make a it a little easier.

```ruby
test :refresh_without_scope_parameter
   metadata do
      name 'Server successfully refreshes the access token when optional scope parameter ommited.'
      id '03'
      ...
   end
   ....
end
```

This is a little more of a burden for the test writer, because you need to come up with two good "names", instead of just one.  And you need to make sure they stay in sync if you change the name (though I would hope that the key is succint enough that it wouldn't ever change unless in the case of a major refactor, while I could imagine the longer "name" could be rephrased).

Also this seems a bit redundant with the 'test id', which is enforced to be a number in a certain format.  Though we acknowledge that test ids are much more likely to change while tests are *under development*, if you add a test in the middle then all the test ids change.  Note that after a test procedure is finalized they can't change because that defeats their purpose, at that point we'll need to allow things like '03b' if we want to add another test in the middle, or change the test substantially.

Another option is to just keep this dsl the same, and in our self-tests just reference the full name.  But if we tweak the wording we'll have to update the self-test...

Reply:
I like this suggestion. Using a symbol rather than a string signals that it's an identifier, `BIG_CONSTANTS_LIKE_THIS` are annoying, and this separates the identification of a test from display concerns.

It is a bit redundant with the id, but I think the current ids work well for the UI where you have `SD-02: Configuration from well-known endpoint contains required fields`, but work less well for tests where you have to look at the sequence to know which test`Sequence['SD-02']` really is (in addition to the reordering problem).